lib/orderbook/MatchingEngine.ts

```diff
@@ -1,23 +1,58 @@
 import assert from 'assert';
 import FastPriorityQueue from 'fastpriorityqueue';
-import { matchingEngine } from '../types';
+import { matchingEngine, orders } from '../types';
 import { OrderingDirection } from '../types/enums';
 import Logger from '../Logger';
-import { StampedOrder, StampedOwnOrder, StampedPeerOrder } from '../types/orders';
+import { isOwnOrder, StampedOrder, StampedOwnOrder, StampedPeerOrder } from '../types/orders';
 
 type SplitOrder = {
   matched: StampedOrder;
   remaining: StampedOrder;
 };
 
+type orderId = string;
+
+type OrderList<T extends orders.StampedOrder> = Map<orderId, T>;
+
+type OrderSidesLists<T extends orders.StampedOrder> = {
+  buy: OrderList<T>,
+  sell: OrderList<T>,
+};
+
+type OrderSidesArrays<T extends orders.StampedOrder> = {
+  buy: T[],
+  sell: T[],
+};
+
+type OrderSidesQueues = {
+  buy: FastPriorityQueue<StampedOrder>,
+  sell: FastPriorityQueue<StampedOrder>,
+};
+
 /** A class to represent a matching engine responsible for matching orders for a given trading pair according to their price and quantity. */
 class MatchingEngine {
-  public buyOrders: FastPriorityQueue<StampedOrder>;
-  public sellOrders: FastPriorityQueue<StampedOrder>;
+  /** Active buy/sell orders priority queue. */
+  public queues: OrderSidesQueues;
+  /** A map between active own buy/sell order ids and orders. */
+  public ownOrders: OrderSidesLists<orders.StampedOwnOrder>;
+  /** A map between active peer buy/sell order ids and orders. */
```

Comment:
I think as long as we keep this `OrderSides` structure we should clarify in the comments here that these are a pair of maps rather than a single map.